contributors/devel/development.md

```diff
@@ -160,6 +154,97 @@ images.
+The best way to validate your current setup is to build a small part of Kubernetes. This way you can address issues without waiting for the full build to complete. To build a specific part of Kubernetes use the `WHAT` environment variable to let the build scripts know you want to build only a certain package/executable.
+
+```sh
+make WHAT=cmd/{$package_you_want}
+```
+
+*Note:* This applies to all top level folders under kubernetes/cmd.
+
+So for the cli, you can run:
+
+```sh
+make WHAT=cmd/kubectl
+```
+
+If everything checks out you will have an executable in the `_output/bin` directory to play around with.
+
+*Note:* If you are using `CDPATH`, you must either start it with a leading colon, or unset the variable. The make rules and scripts to build require the current directory to come first on the CD search path in order to properly navigate between directories.
+
+```sh
+cd $working_dir/kubernetes
+make
+```
+
+To remove the limit on the number of errors the Go compiler reports (default
+limit is 10 errors):
+```sh
+make GOGCFLAGS="-e"
+```
+
+To build with optimizations disabled (enables use of source debug tools):
+
+```sh
+make GOGCFLAGS="-N -l"
+```
+
+To build binaries for all platforms:
+
+```sh
+make cross
+```
+
+#### Install etcd
+
+```sh
+cd $working_dir/kubernetes
+
+# Installs in ./third_party/etcd
+hack/install-etcd.sh
+
+# Add to PATH
+echo export PATH="\$PATH:$working_dir/kubernetes/third_party/etcd" >> ~/.profile
+```
+
+#### Test
+
+```sh
+cd $working_dir/kubernetes
+
+# Run all the presubmission verification. Then, run a specific update script (hack/update-*.sh)
+# for each failed verification. For example:
+#   hack/update-gofmt.sh (to make sure all files are correctly formatted, usually needed when you add new files)
+#   hack/update-bazel.sh (to update bazel build related files, usually needed when you add or remove imports)
+make verify
+
+# Alternatively, run all update scripts to avoid fixing verification failures one by one.
+make update
+
+# Run every unit test
+make test
+
+# Run package tests verbosely
+make test WHAT=./pkg/api/helper GOFLAGS=-v
+
+# Run integration tests, requires etcd
+# For more info, visit https://git.k8s.io/community/contributors/devel/sig-testing/testing.md#integration-tests
+make test-integration
+
+# Run e2e tests by building test binaries, turn up a test cluster, run all tests, and tear the cluster down
+# Equivalent to: go run hack/e2e.go -- -v --build --up --test --down
+# Note: running all e2e tests takes a LONG time! To run specific e2e tests, visit:
+# https://git.k8s.io/community/contributors/devel/sig-testing/e2e-tests.md#building-kubernetes-and-running-the-tests
+make test-e2e
+```
+
+See the [testing guide](/contributors/devel/sig-testing/e2e-tests.md) and [end-to-end tests](/contributors/devel/sig-testing/e2e-tests.md)
```

Comment:
I think you changed the wrong link here 😅 

the testing guide is `testing.md`, please change that back?

and both links should be just `./testing.md` and `./e2e-tests.md`.